### PR TITLE
Pubmatic: Making Adslot param optional

### PR DESCRIPTION
**Pull request by pm-harshad-mane**
## Type of change
- [X] Feature

## Description of change
With the addition of default site & ad tag support in at PubMatic Openbid, it's no longer a mandatory requirement to send adslot name parameter.

New behavior:
- adSlot name should be optional
- If adSlot name is not present, we should send the first size as w and h, and other sizes in the format array

### CODE REVIEW

**Comment by pm-harshad-mane** [comment]
Hello @jsnellbaker , @jaiminpanchal27 
Can you please review the PR?

**Comment by jaiminpanchal27** [comment]
@pm-harshad-mane Have assigned this to myself. I will review soon

File: modules/pubmaticBidAdapter.js [hunk 1]
```diff
@@ -1002,7 +992,7 @@ export const spec = {
   transformBidParams: function (params, isOpenRtb) {
     return utils.convertTypes({
       'publisherId': 'string',
-      'adSlot': 'string'
+      'adSlot': 'string' // TODO: do we need to change here as well?
```
Comment (jaiminpanchal27) [inline modules/pubmaticBidAdapter.js]: No this is fine as it is. Type will be converted if it exists

File: test/spec/modules/pubmaticBidAdapter_spec.js [hunk 2]
```diff
@@ -668,28 +668,41 @@ describe('PubMatic adapter', function () {
...
-              adSlot: 15671365
+              publisherId: '301'
             }
           },
           isValid = spec.isBidRequestValid(validBid);
-        expect(isValid).to.equal(false);
+        expect(isValid).to.equal(true);
       });
+
+      // commenting following test-case as we are making param adSlot optional
```
[the remark this answers is not in the document]
Comment (pm-harshad-mane) [reply test/spec/modules/pubmaticBidAdapter_spec.js]: Deleted commented test cases

File: modules/pubmaticBidAdapter.js (hunk 1 above)
Comment (pm-harshad-mane) [reply modules/pubmaticBidAdapter.js]: removed the todo comment

**Comment by bretg** [comment]
@jaiminpanchal27 - ready to merge?

**Comment by bretg** [comment]
looks to me like the requested changes were made. Merging.

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/1299

**Comment by pm-harshad-mane** [comment]
Thank you @bretg , @jaiminpanchal27

**Pull request merged**